src/shiv/bootstrap/__init__.py

```diff
@@ -83,6 +83,12 @@ def extract_site_packages(archive, target_path):
     shutil.move(str(target_path_tmp), str(target_path))
 
 
+def _first_sitedir_index():
+    for index, part in enumerate(sys.path):
+        if Path(part).stem == 'site-packages':
+            return index
+
```

Comment:
This function returns `None` if it doesn't find any `site-packages` and it returns an `int` if it does find some. I'd prefer the function to return always an `int` for easier type checking. Why not `return len(sys.path)` at the bottom outside the `for` loop. That would solve the typing consistency issue and would make an `or len(sys.path)` unnecessary on the line 103 below.

Reply:
Personally, I prefer the return type to be `Optional[int]`, as we're looking for something specific (first sitedir index) and if not found, returning `None` is logical since no sitedir was was present.